tests/unit_tests/test_game.py

```diff
@@ -1074,6 +1074,158 @@ def custom_game(event_loop, database, game_service, game_stats_service):
     # No guarantees on scores for conflicting results.
 
 
+async def test_single_wrong_report_still_rated_correctly(game: Game, player_factory):
+    # based on replay with UID 11255492
+    game.state = GameState.LOBBY
+    players = [
+            (player_factory(
+                login=f"{i}",
+                player_id=i,
+                global_rating=rating,
+                with_lobby_connection=False
+            ), score, team)
+            for i, (rating, score, team) in enumerate([
+                (Rating(1500, 250), 0, 2),
+                (Rating(1500, 250), 0, 2),
+                (Rating(1500, 250), 0, 2),
+                (Rating(1500, 250), 0, 2),
+                (Rating(1500, 250), 0, 2),
+                (Rating(1500, 250), 0, 2),
+                (Rating(1500, 250), 0, 3),
+                (Rating(1500, 250), 0, 3),
+                (Rating(1500, 250), 0, 3),
+                (Rating(1500, 250), 0, 3),
+                (Rating(1500, 250), 0, 3),
+                (Rating(1500, 250), 0, 3),
+            ], 1)]
+    add_connected_players(game, [player for player, _, _ in players])
+    for player, _, team in players:
+        game.set_player_option(player.id, 'Team', team)
+        game.set_player_option(player.id, 'Army', player.id - 1)
+    await game.launch()
+
+    await game.add_result(players[0][0].id, 1, "defeat", 0)
```

Comment:
Here's an idea. How about we extract this to a json file that we can put into `tests/data`. We could even write a script to create such json files from parsing actual server log snippets. Then we could easily add lots of real test cases without manually entering the results like this.

Maybe some simple format like:
```
{
    "teams": {
        2: [1,2,3,4,5,6],
        3: [7,8,9,10,11,12]
    },
    "results": [
        [3, 1, "defeat", 0],
        [9, 1, "defeat", 0],
        ...
    ],
    "winning_team": 2
}
```
We would have to add the `winning_team` field manually to each file.